### PR TITLE
Fix botched py3 compat change

### PR DESCRIPTION
**Pull request by jacksontj**

### CODE REVIEW

File: salt/client/mixins.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def wrapper(*args, **kwargs):
                    }
             pub_data = {}
             # pull out pub_data if you have it
-            for k, v in six.iteritems(list(kwargs.items())):
+            for k, v in six.iteritems(dict(kwargs)):
```
Comment (s0undt3ch) [inline salt/client/mixins.py]: If kwargs is already a dict there's no need to cast it. Any other reason?

Comment (jacksontj) [inline salt/client/mixins.py]: To make a copy of the dict, before this was doing .items() which iterates over a *copy* of the dict

Comment (s0undt3ch) [inline salt/client/mixins.py]: Yeah. Makes sense. Didn't saw the comment above when looking at the diff. The dict is being changed while iterating.

Ignore me :)

**Comment by thatch45** [comment]
Go Go Jenkins!

**Pull request merged**